app/services/orders/check_stock_service.rb

```diff
@@ -4,12 +4,24 @@ module Orders
   class CheckStockService
     attr_reader :order
 
-    def initialize(order: )
+    def initialize(order:)
       @order = order
     end
 
     def sufficient_stock?
       @sufficient_stock ||= order.insufficient_stock_lines.blank?
     end
+
+    def update_line_items
+      return [] if sufficient_stock?
+
+      variants = []
+      order.insufficient_stock_lines.each do |line_item|
+        order.contents.update_item(line_item, { quantity: line_item.variant.on_hand })
+        variants.push line_item.variant
+      end
+
+      variants
```

Comment:
I think this could be reduced down to a `map`, ie:

```suggestion
      order.insufficient_stock_lines.map do |line_item|
        order.contents.update_item(line_item, { quantity: line_item.variant.on_hand })
        line_item.variant
      end
```

Although it's a little confusing the `update_line_items` returns a list of Variants. One thought I had was to just return the LineItem, then in the calling code run another map to fetch the variants. But then that's two loops.. so this is fine.